envisat_format/include/envisat_utils.h

```diff
@@ -51,4 +61,10 @@ inline boost::posix_time::ptime StrToPtime(std::string str) {
         exit(1);
     }
     return time;
-}
\ No newline at end of file
+}
+
+inline mjd MjdAddMicros(const mjd& d, std::decay<decltype(std::declval<mjd>().micros)>::type micros_to_add) {
```

Comment:
??? This makes no sense, various template programming utilities used in here for a non-template function?